### PR TITLE
Fix data check on GVRMesh.setVecXVector(), also put check on all setX().

### PR DESCRIPTION
**Pull request by chenchao1407**
Add this exception mechanism to make sure it won't crash at native level without any useful information or call stack.

GearVRf-DCO-1.0-Signed-off-by: Chao Chen
c.chen@samsung.com

### CODE REVIEW

**Comment by chenchao1407** [comment]
Hi, Jon.

I modified the setVecXVector based on the Mihail's comment on my previous one.

You can see that in mesh_jni.cpp, the logic of setVec2Vector is as the following:
    std::vector<glm::vec2> native_vec2_vector;
    for (int i = 0; i < vec2_vector_length; ++i) {
        native_vec2_vector.push_back(vec2_vector_pointer[i]);
    }

It is similar in other setVecXVector, so that this is why I check the input like that.

**Comment by JonShemitz** [comment]
Yes. So your test will keep that code from breaking. But what happens if you pass an array 4 floats (two vec2-s) to a mesh with 64 vertices?

**Comment by chenchao1407** [comment]
And also, the reason why I use switch-case is mostly for null pointer check. Otherwise we need to do another null pointer check as you point out.

I can use the mechanism you indicate. But I think the logic of this patch should be fine.

**Comment by chenchao1407** [comment]
OK. I got your point. I am going to modify the codes and update this pull request.

**Comment by chenchao1407** [comment]
Pull request updated.

**Comment by chenchao1407** [comment]
Hi Jon
This is updated based on your comments. Can you help check again?

Thanks in advance.

File: GVRf/Framework/src/org/gearvrf/GVRMesh.java [hunk 1]
```diff
@@ -272,13 +291,31 @@ public GVRMesh getBoundingBox() {
...
+            throw Exceptions
+                    .IllegalArgument(
+                            "The input array should be an array of %d-component elements whose length is non-zero and divisible by %d. But current data length is %d.",
+                            expectedComponents, expectedComponents, dataLength);
+        }
+    }
+
+    private void verticesCheck(int dataLength, int expectedComponents) {
+        int verticesNumber = getVertices().length / 3;
+        int numberOfElements = dataLength / expectedComponents;
```
Comment (JonShemitz) [inline GVRf/Framework/src/org/gearvrf/GVRMesh.java]: What if there are 12 vertices; dataLength is 37; and expectedComponents is 3?

**Comment by JonShemitz** [comment]
This looks pretty good. The error check is not 100% accurate but I'm not sure that's 100% necessary ...

File: GVRf/Framework/src/org/gearvrf/GVRMesh.java (hunk 1 above)
Comment (chenchao1407) [inline GVRf/Framework/src/org/gearvrf/GVRMesh.java]: Hi Jon.

I think for this issue, the isValidData has already taken care of that so it should be fine.

**Comment by JonShemitz** [comment]
Yes, thanks - I missed the way setValidData() has already done the modulus check.

**Pull request merged**